### PR TITLE
[1ES] Added 1ES pipeline files

### PR DESCRIPTION
**Pull request by surgupta-msft**
This PR introduces new CI files for a new build pipeline.

### CODE REVIEW

File: eng/ci/templates/jobs/build-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      arguments: '-c $(buildConfiguration)'
+      projects: 'Azure.Web.DataProtection.sln'
+
+  - task: DotNetCoreCLI@2
+    displayName: 'Run tests'
+    inputs:
+      command: 'test'
+      arguments: '--no-build -c $(buildConfiguration)'
+      projects: |
+        test/**/**Tests.csproj
```
Comment (jviau) [inline eng/ci/templates/jobs/build-artifacts.yml:45]: If `Azure.Web.DataProtection.sln` includes the test projects, you can use that here. `dotnet test` will be able to use that solution to find test projects and ignore non test projects.

File: eng/ci/templates/jobs/build-artifacts.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  - ${{ if eq(parameters.SkipBuildArtifacts, false) }}:
+    - task: DotNetCoreCLI@2
+      displayName: 'Build package'
+      inputs:
+        command: 'custom'
+        custom: 'pack'
+        arguments: '-c Release -p:Version="$(versionPrefix).$(buildVersion)-$(versionSuffix)" -o packages'
```
Comment (jviau) [inline eng/ci/templates/jobs/build-artifacts.yml:53]: Add `--no-build`

**Review by jviau — CHANGES_REQUESTED** [review]

File: eng/ci/templates/jobs/build-artifacts.yml (hunk 1 above)
Comment (surgupta-msft) [reply eng/ci/templates/jobs/build-artifacts.yml]: Done

File: eng/ci/templates/jobs/build-artifacts.yml (hunk 2 above)
Comment (surgupta-msft) [reply eng/ci/templates/jobs/build-artifacts.yml]: Done

**Review by jviau — APPROVED** [review]

**Pull request merged**